code/game/objects/items/extinguisher.dm

```diff
@@ -143,64 +143,80 @@
 		var/direction = get_dir(src,target)
 
 		if(user.buckled && isobj(user.buckled) && !user.buckled.anchored)
-			spawn(0)
-				var/obj/B = user.buckled
-				var/movementdirection = turn(direction,180)
-				step(B, movementdirection)
-				sleep(1)
-				step(B, movementdirection)
-				sleep(1)
-				step(B, movementdirection)
-				sleep(1)
-				step(B, movementdirection)
-				sleep(2)
-				step(B, movementdirection)
-				sleep(2)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
+			var/obj/B = user.buckled
+			var/movementdirection = turn(direction,180)
+			addtimer(CALLBACK(src, /obj/item/extinguisher/.proc/move_chair, B, movementdirection, 0), 1)

```

Comment:
`/.proc/`?

Reply:
That's how I saw in the code and it works. If there's a better variant, feel free to drop me some advice.

Reply:
I'm currently testing it without the . to see if it would work aswell. I had some problems getting it to work before, so I opted for the secure way.